### ISSUE TITLE
The GC_Poll Create/Insert logic appears to be flawed

### ISSUE THREAD

**Issue by AaronRobinsonMSFT** [dotnet/coreclr#27186]
The code responsible for creating and inserting GC_Poll instances appears to have degraded since porting over from .NET Framework.

https://github.com/dotnet/coreclr/blob/45a136968ab6e0446f5309852008b3cf151f83a6/src/jit/flowgraph.cpp#L3574

Calling the above function when either `GCPOLL_CALL` pr `GCPOLL_INLINE` is set results in multiple asserts being fired. This was discovered during the suppress GC work - https://github.com/dotnet/coreclr/pull/26458.

/cc @briansull @dotnet/jit-contrib

**Comment by jkotas** [comment]
The SuppressGCTransitionAttribute optimization leaves performance on the table because of this issue.

**Comment by briansull** [comment]
This will analyze loops to prove that a managed call always occurs in the execution of one iteration.
If it proves this it will omit the GC_POLL call for the loop.
When we discussed this behavior during #26458, you advocated for always calling the GC_POLL rather than optimizing it away.

**Comment by jkotas** [comment]
How you recommend that we implement the inlined GC poll for SuppressGCTransitionAttribute? Ignore what is there today and just do it from scratch?

**Comment by AaronRobinsonMSFT** [comment]
@briansull The issue here was that during some scenarios no assert was hit with `GCPOLL_INLINE`, but there were cases when it would assert. Asserts also occurred when setting `GCPOLL_CALL`. There are obviously multiple scenarios here.

Inline PInvoke in an inlineable function.

``` csharp
   public static int GetTime()
   {
       return InlineablePInvokeWithGCSuppression();
   }
```

PInvoke in non-inlined function

``` csharp
   public static int GetAccTime()
   {
       int acc = 0;
       foreach (...)
       {
          acc += InlineablePInvokeWithGCSuppression();
       }

       return acc;
   }
```

Then there are scenarios involving CrossGen which can't use `GCPOLL_INLINE` at the moment, but requires `GCPOLL_CALL` - see https://github.com/dotnet/coreclr/issues/27206.

**Comment by briansull** [comment]
> How you recommend that we implement the inlined GC poll for SuppressGCTransitionAttribute? 

Yes, We should fix the codepath that we use for SuppressGCTransitionAttribute to support GCPOLL_INLINE in all cases.

The existing code fgCreateGCPolls()  was last used to support the MacOS during the pre-RyuJIT time frame.  That code should be deleted as it tries to eliminate the GCPOLL call when it can prove that a managed call will be made in every loop iteration, which is not what we want with this new feature.
